### PR TITLE
Adding Support for HTTP 2 APNS authentication

### PR DESCRIPTION
**Pull request by amolr**
1. Added Support for HTTP 2 APNS authentication and updated tests
2. Added a checkNameavailability API to include new request/response parameters. Will add tests once this is in prod

### CODE REVIEW

**Comment by balajikris** [comment]
Can you please provide the following information:
1. Except for gulpfile, package.json and all tests, everything is auto-generated by the latest version of autorest - correct? [changes to gulp, package.json and tests are hand written]
2. cleanup/squash your commits into two commits - one with all the product changes and another with just the tests. In it current state, it messes history with adding and deleting temp files like `npm-debug.log`. Please squash commits and help maintain git history.

**Comment by amolr** [comment]
Yes only gulp file, package.json and tests are hand written.
Also, i have squashed the commits into product and test related ones

File: nodeSDK.njsproj [hunk 1]
```diff
@@ -16,6 +16,13 @@
     <EnableTypeScript>true</EnableTypeScript>
     <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">11.0</VisualStudioVersion>
     <VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>
+    <StartWebBrowser>False</StartWebBrowser>
+    <Environment>AZURE_SUBSCRIPTION_ID=2cac2a14-ba6b-46a6-bce8-2d9781a41ba2
+CLIENT_ID=0e44d72c-4a92-4fd0-a640-d165dde5da87
+DOMAIN=microsoft.com
+APPLICATION_SECRET=KsQw6kI7QKLair8uZkqFkMblh2ryIkJB98usb6xr2wU=
```
Comment (balajikris) [inline nodeSDK.njsproj]: I don't think you need any edits to the ".proj" file to make tests pass. I assume these changes are for your local debugging. If that's the case remove this file from the changes.

Comment (balajikris) [inline nodeSDK.njsproj]: Otherwise - things are good.

File: nodeSDK.njsproj [hunk 2]
```diff
@@ -1532,8 +1539,12 @@
       <TestFramework>
       </TestFramework>
     </Compile>
-    <Compile Include="test\services\notificationHubsManagement\namespace-tests.js" />
-    <Compile Include="test\services\notificationHubsManagement\notificationHub-tests.js" />
+    <Compile Include="test\services\notificationHubsManagement\namespace-tests.js">
+      <TestFramework>Mocha</TestFramework>
+    </Compile>
+    <Compile Include="test\services\notificationHubsManagement\notificationHub-tests.js">
+      <TestFramework>Mocha</TestFramework>
```
Comment (balajikris) [inline nodeSDK.njsproj]: Not required either. Simply for your local debugging.

**Comment by amolr** [comment]
The required changes for the proj files has been made

**Pull request merged**